beacon_node/network/src/beacon_processor/mod.rs

```diff
@@ -566,16 +566,12 @@ impl<T: BeaconChainTypes> WorkEvent<T> {
 impl<T: BeaconChainTypes> std::convert::From<ReadyWork<T>> for WorkEvent<T> {
     fn from(ready_work: ReadyWork<T>) -> Self {
         match ready_work {
-            ReadyWork::Block(QueuedBlock {
-                peer_id,
-                block,
-                seen_timestamp,
-            }) => Self {
+            ReadyWork::Block(queued) => Self {
                 drop_during_sync: false,
                 work: Work::DelayedImportBlock {
-                    peer_id,
-                    block: Box::new(block),
-                    seen_timestamp,
+                    peer_id: queued.peer_id,
+                    block: Box::new(queued.block),
+                    seen_timestamp: queued.seen_timestamp,
```

Comment:
If you box just the block, this does not need to change

Reply:
Better explanation: Boxing the `GossipVerifiedBlock` [here](https://github.com/sigp/lighthouse/blob/f3c237cfa0ca30e09492b42db44c855c2a7fc088/beacon_node/network/src/beacon_processor/work_reprocessing_queue.rs#L91-L96) would make it so that the queued block does not need to be re boxed so many times, since the large part of the type is just that one. You'll note that by changing this this way, most of the new changes related to that are not needed

Reply:
Ah yep, it's way better: implemented here: https://github.com/sigp/lighthouse/pull/2850/commits/b737f16662f8e53195fb04060105d6f9f4227f4b

Think there's an EF test failure because I renamed a field we needed for deserialization